packages/react-instantsearch-dom/src/hooks/useAnswers.js

```diff
@@ -0,0 +1,99 @@
+import React, { useState, useEffect, useMemo, useContext } from 'react';
+import { instantSearchContext } from 'react-instantsearch-core';
+import { createConcurrentSafePromise } from '../lib/createConcurrentSafePromise';
+import { debounce } from '../lib/debounce';
+
+function hasReactHooks() {
+  // >= 16.8.0
+  const [major, minor] = React.version.split('.').map(Number);
+  return major >= 17 || (major === 16 && minor >= 8);
+}
+
+export default function useAnswers({
+  searchClient,
+  queryLanguages,
+  attributesForPrediction,
+  nbHits,
+  renderDebounceTime = 100,
+  searchDebounceTime = 100,
+  ...extraParameters
+}) {
+  if (!hasReactHooks()) {
+    throw new Error(
+      `\`Answers\` component and \`useAnswers\` hook require all React packages to be 16.8.0 or higher.`
+    );
+  }
+  const context = useContext(instantSearchContext);
+  const [query, setQuery] = useState();
+  const [index, setIndex] = useState();
+  const [isLoading, setIsLoading] = useState();
+  const [hits, setHits] = useState();
```

Comment:
We should default this to `[]` to not introduce another type and conditional in user code:

![image](https://user-images.githubusercontent.com/6137112/106026692-6dfeaf80-60ca-11eb-95cb-059a9060524f.png)

```suggestion
  const [hits, setHits] = useState([]);
```

Reply:
It makes sense https://github.com/algolia/react-instantsearch/pull/2996/commits/0228c51c2d65e477fbce6471e04a85597f5bfda6